src/common/transformations/src/transformations/common_optimizations/eliminate_loop_inputs_outputs.cpp

```diff
@@ -0,0 +1,185 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/eliminate_loop_inputs_outputs.hpp"
+
+#include <unordered_map>
+
+#include "itt.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/core/type.hpp"
+#include "openvino/op/loop.hpp"
+#include "openvino/op/parameter.hpp"
+#include "openvino/op/result.hpp"
+#include "openvino/op/tensor_iterator.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+
+using namespace std;
+using namespace ov::element;
+using namespace ov::pass::pattern;
+
+namespace {
+std::shared_ptr<ov::op::v0::Parameter> get_parent_param(const std::shared_ptr<ov::op::v0::Result>& result) {
+    return ov::as_type_ptr<ov::op::v0::Parameter>(result->input_values()[0].get_node_shared_ptr());
+}
+}  // namespace
+
+ov::pass::EliminateLoopInputsOutputs::EliminateLoopInputsOutputs() {
+    MATCHER_SCOPE(EliminateLoopInputsOutputs);
+
+    auto subgraph_label = wrap_type<ov::op::v5::Loop, ov::op::v0::TensorIterator>();
+
+    matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        using namespace ov::op::util;
+        using InvariantD = MultiSubGraphOp::InvariantInputDescription;
+        using SlicedD = MultiSubGraphOp::SliceInputDescription;
+        using MergedD = MultiSubGraphOp::MergedInputDescription;
+        using OutputD = MultiSubGraphOp::BodyOutputDescription;
+        using ConcatD = MultiSubGraphOp::ConcatOutputDescription;
+
+        // deleting parameter/results are inserted to the beginning of deque, adding Descriptions to the end
```

Comment:
the comment is no longer relevant , please delete

Reply:
fixed.